File: internal/domain/oauth_account.go
package domain

import "time"

type OAuthAccount struct {
	ID             int64      `json:"id"`
	UserID         int64      `json:"user_id"`
	Provider       string     `json:"provider"`
	ProviderUserID string     `json:"provider_user_id"`
	Email          *string    `json:"email,omitempty"`
	Name           *string    `json:"name,omitempty"`
	AvatarURL      *string    `json:"avatar_url,omitempty"`
	AccessToken    *string    `json:"-"` // Never expose in JSON
	RefreshToken   *string    `json:"-"` // Never expose in JSON
	TokenExpiresAt *time.Time `json:"token_expires_at,omitempty"`
	CreatedAt      *time.Time `json:"created_at"`
	UpdatedAt      *time.Time `json:"updated_at"`
}

type CreateOAuthAccountAction struct {
	UserID         int64
	Provider       string
	ProviderUserID string
	Email          *string
	Name           *string
	AvatarURL      *string
	AccessToken    *string
	RefreshToken   *string
	TokenExpiresAt *time.Time
}

type UpdateOAuthAccountAction struct {
	ID             int64
	UserID         int64
	Email          *string
	Name           *string
	AvatarURL      *string
	AccessToken    *string
	RefreshToken   *string
	TokenExpiresAt *time.Time
}

type UpdateOAuthTokensAction struct {
	ID             int64
	UserID         int64
	AccessToken    *string
	RefreshToken   *string
	TokenExpiresAt *time.Time
}

const (
	OAuthProviderGoogle  = "google"
	OAuthProviderGitHub  = "github"
	OAuthProviderDiscord = "discord"
	OAuthProviderTwitter = "twitter"
)

File: frontend/src/types/models/oauth_account.ts
export type OAuthAccount = {
	id: number
	user_id: number
	provider: 'google' | 'github' | 'discord' | 'twitter'
	provider_user_id: string
	email?: string
	name?: string
	avatar_url?: string
	token_expires_at?: string
	created_at?: string
	updated_at?: string
}

export const OAuthProviders = {
	GOOGLE: 'google',
	GITHUB: 'github',
	DISCORD: 'discord',
	TWITTER: 'twitter',
} as const

export type OAuthProvider = (typeof OAuthProviders)[keyof typeof OAuthProviders]

export type OAuthAuthResponse = {
	auth_url: string
	state: string
}
